packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -426,17 +511,45 @@ class CupertinoPageTransition extends StatelessWidget {
   final Animation<Offset> _secondaryPositionAnimation;
   final Animation<Decoration> _primaryShadowAnimation;
 
+  /// Animation
+  final Animation<double> secondaryRouteAnimation;
+
   /// The widget below this widget in the tree.
   final Widget child;
 
-  @override
-  Widget build(BuildContext context) {
+  /// The delegated transition.
+  static Widget delegateTransition(BuildContext context, Widget? child, Animation<double> secondaryAnimation) {
+    final Animation<Offset> delegatedPositionAnimation =
+      CurvedAnimation(
+        parent: secondaryAnimation,
+        curve: Curves.linearToEaseOut,
+        reverseCurve: Curves.easeInToLinear,
+      ).drive(_kMiddleLeftTween);
     assert(debugCheckHasDirectionality(context));
     final TextDirection textDirection = Directionality.of(context);
     return SlideTransition(
-      position: _secondaryPositionAnimation,
+      position: delegatedPositionAnimation,
       textDirection: textDirection,
       transformHitTests: false,
+      child: child,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasDirectionality(context));
+    final TextDirection textDirection = Directionality.of(context);
+    return DelegatedTransition(
```

Comment:
As we talked about, it's tricky to have to put DelegatedTransition in every existing page transition in order to get MBS support. Users won't realize they have to do that in their custom transitions. Also, it's not possible to determine whether or not a certain transition/route supports MBS or not.

Instead of putting it in every transition, maybe you could add it somewhere higher like MaterialPageRoute.buildTransitions (that's where it currently gets the transition from PageTransitionsTheme). Then support for MBS is decided at the route instead of at the transition, which makes more sense.

Reply:
+1 to this. This seems like a bit of a foot gun for defining your own routing subclasses.